From a00bd3b1d347ef40d9f7fa5c66f97c79c95efaa5 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Fri, 14 May 2021 16:48:10 +0800
Subject: [PATCH] dts: npcm845 dts update

1. modify the chip select info under FIU0
2. remove spi-rx-bus-width in FIU0/FIU3

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 .../boot/dts/nuvoton/nuvoton-npcm845-evb.dts  | 69 +++++++------------
 1 file changed, 25 insertions(+), 44 deletions(-)

diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
index 6956fd38d9cf..07343e87ea69 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
@@ -147,59 +147,41 @@
 
 		fiu0: spi@fb000000 {
 			status = "okay";
-			spi-nor@0 {
+			spi-nor@1 {
 				compatible = "jedec,spi-nor";
 				#address-cells = <1>;
 				#size-cells = <1>;
-				spi-rx-bus-width = <2>;
-				reg = <0>;
+				reg = <1>;
 				spi-max-frequency = <5000000>;
-				partitions@80000000 {
+
+				partitions@88000000 {
 					compatible = "fixed-partitions";
 					#address-cells = <1>;
 					#size-cells = <1>;
-					bbuboot1@0 {
-						label = "bb-uboot-1";
-						reg = <0x0000000 0x80000>;
-						read-only;
-						};
-					bbuboot2@80000 {
-						label = "bb-uboot-2";
-						reg = <0x0080000 0x80000>;
-						read-only;
-						};
-					envparam@100000 {
-						label = "env-param";
-						reg = <0x0100000 0x40000>;
+					bmc@0{
+						label = "bmc";
+						reg = <0x000000 0x2000000>;
+					};
+					u-boot@0 {
+						label = "u-boot";
+						reg = <0x0000000 0xC0000>;
 						read-only;
-						};
-					spare@140000 {
-						label = "spare";
-						reg = <0x0140000 0xC0000>;
-						};
+					};
+					u-boot-env@100000{
+						label = "u-boot-env";
+						reg = <0x00100000 0x40000>;
+					};
 					kernel@200000 {
 						label = "kernel";
-						reg = <0x0200000 0x400000>;
-						};
-					rootfs@600000 {
-						label = "rootfs";
-						reg = <0x0600000 0x700000>;
-						};
-					spare1@D00000 {
-						label = "spare1";
-						reg = <0x0D00000 0x200000>;
-						};
-					spare2@0F00000 {
-						label = "spare2";
-						reg = <0x0F00000 0x200000>;
-						};
-					spare3@1100000 {
-						label = "spare3";
-						reg = <0x1100000 0x200000>;
-						};
-					spare4@1300000 {
-						label = "spare4";
-						reg = <0x1300000 0x0>;
+						reg = <0x0200000 0x800000>;
+					};
+					rofs@800000 {
+						label = "rofs";
+						reg = <0xa00000 0x1500000>;
+					};
+					rwfs@1f00000 {
+						label = "rwfs";
+						reg = <0x1f00000 0x100000>;
 					};
 				};
 			};
@@ -234,7 +216,6 @@
 				compatible = "jedec,spi-nor";
 				#address-cells = <1>;
 				#size-cells = <1>;
-				spi-rx-bus-width = <2>;
 				reg = <0>;
 				spi-max-frequency = <5000000>;
 				partitions@A0000000 {
-- 
2.17.1

